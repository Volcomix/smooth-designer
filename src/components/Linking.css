.Linking {
  position: absolute;
  top: 0;
  left: 0;
  width: 1px;
  height: 1px;
  overflow: visible;
  z-index: 15;
  pointer-events: none;
}

.Linking-link {
  stroke: var(--primary1-color);
}
