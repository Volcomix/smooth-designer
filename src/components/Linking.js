//@flow
import React from 'react'
import './Linking.css'
import type { Block } from '../types'

export type Props = {
  fromBlock: Block,
  toMouse: { x: number, y: number },
}

const Linking = ({ fromBlock, toMouse }: Props) => (
  <svg className="Linking">
    <line
      className="Linking-link"
      x1={fromBlock.x}
      y1={fromBlock.y}
      x2={toMouse.x}
      y2={toMouse.y}
    />
  </svg>
)

export default Linking
