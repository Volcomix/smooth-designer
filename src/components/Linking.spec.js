//@flow
import React from 'react'
import { shallow } from 'enzyme'
import Linking from './Linking'

it('renders without crashing', () => {
  const props = {
    fromBlock: { id: '0', name: 'Block', x: 10, y: 20, width: 0, height: 0 },
    toMouse: { x: 30, y: 40 },
  }
  const wrapper = shallow(<Linking {...props} />)
  expect(wrapper).toMatchSnapshot()
})
