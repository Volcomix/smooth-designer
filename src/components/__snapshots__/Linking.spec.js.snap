// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<svg
  className="Linking"
>
  <line
    className="Linking-link"
    x1={10}
    x2={30}
    y1={20}
    y2={40}
  />
</svg>
`;
